### PR TITLE
Adjustments to E2E tests to decrease CI test run time

### PR DESCRIPTION
**Pull request by avalluri**
- skip driver late-binding stress tests in direct mode - supposed to save ~40m
- limit running package manager and imagefile unit tests part of E2E tests - supposed to save ~10m

### CODE REVIEW

**Review by pohly — APPROVED** [review]

**Comment by okartau** [comment]
Did (and how) we verify tests actually run in shorter time with these changes? 
The job for this PR took 7h57min which is significantly longer than other PRs tested close in time (879, 880 about 5h30min)
https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/job/pmem-csi/view/change-requests/

**Comment by okartau** [comment]
I hope that Jenkins duration display is not affected by some side effect, like including wait-time for workers. Around running these jobs, worker capacity as maxed and some jobs had to wait for some time to get started.

**Comment by avalluri** [comment]
> Did (and how) we verify tests actually run in shorter time with these changes?

Not really, as the disabled/Skipped tests are "[Slow]" tests which do not run on PRs .

> The job for this PR took 7h57min which is significantly longer than other PRs tested close in time (879, 880 about 5h30min)
> https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/job/pmem-csi/view/change-requests/

This is bit strange, This PR should skip existing tests, but not introduce any delays or run new tests.

**Comment by okartau** [comment]
> Not really, as the disabled/Skipped tests are "[Slow]" tests which do not run on PRs

ah, in that case, my checking for run times is not relevant, there has been some other reason, as I suspected.

**Comment by pohly** [comment]
This saved less time than expected. Latest "devel" testing on 1.19 before this change took 8h 33m 28s (https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/devel/315/pipeline/86), with this change 8h 13m 28s (https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/devel/316/pipeline/86).

It's still worthwhile, but we probably can do better.

I noticed that "late binding stress test [Slow] – fedora-1_18.operator-direct-production.driver" still ran. Should that have been skipped?

File: test/test.make [hunk 1]
```diff
@@ -152,6 +152,10 @@ TEST_E2E_SKIP_ALL += should.access.volume.from.different.nodes
 # Test is flawed and will become optional soon (probably csi-test 3.2.0): https://github.com/kubernetes-csi/csi-test/pull/258
 TEST_E2E_SKIP_ALL += NodeUnpublishVolume.*should.fail.when.the.volume.is.missing
 
+# Do not run driver stress tests in direct mode, they are consuming more time(~17m per test)
+# The reason is shredding the ndctl device is consuming most of the time.
+TEST_E2E_SKIP_ALL += direct.*.E2E.late.binding.stress.test
```
Comment (pohly) [inline test/test.make:157]: Including "E2E" makes this not match "operator-lvm-production driver late binding stress test".

**Comment by avalluri** [comment]
> This saved less time than expected. Latest "devel" testing on 1.19 before this change took 8h 33m 28s (https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/devel/315/pipeline/86), with this change 8h 13m 28s (https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/devel/316/pipeline/86).

Could have made ~10min difference if no test failures. Those failed tests took timeout.

File: test/test.make (hunk 1 above)
Comment (avalluri) [reply test/test.make:157]: Yes, I didn't thought about running them through operator.

**Comment by pohly** [comment]
Olev Kartau <notifications@github.com> writes:
> Did (and how) we verify tests actually run in shorter time with these
> changes?

I've looked at recent "devel" builds and drilled down to the actual
execution of the E2E step for 1.19. That runs on the already started
worker and thus should not occur any additional wait time.

**Pull request merged**